src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -154,8 +154,15 @@ def test_input_equals_output(resource_client, input_model, output_model):
     pruned_output_model = prune_properties_if_not_exist_in_path(
         output_model.copy(), pruned_input_model, resource_client.read_only_paths
     )
+
     pruned_output_model = prune_properties_if_not_exist_in_path(
         pruned_output_model, pruned_input_model, resource_client.create_only_paths
     )
-
-    assert pruned_input_model == pruned_output_model
+    # only comparing properties in input model to those in output model and
+    # ignoring extraneous properties that maybe present in output model.
+    assert all(
+        pruned_input_model[key] == pruned_output_model[key]
+        for key in pruned_input_model
```

Comment:
hmm, if output model does not return all properties then we will have a runtime error, which will not surface correctly the issue; we should catch KeyError and raise AssertionError